xma/layers/gru/module.py

```
@@ -73,27 +75,37 @@ def __init__(
-                self.num_forget_input_heads * self.state_head_dim,
-                self.num_reset_input_heads * self.state_head_dim,
-            ),
-            dim=-1,
-        )
+        input, forget_gate, reset_gate = input.split((self.x_shape, self.xf_shape, self.xr_shape), dim=-1)
+
+        if self.conv1d is None:
+            assert conv_state is None
+        else:
+            assert cu_seqlens is None, "depthwise causal conv does not support cu_seqlens"
+            input, conv_state = self.conv1d(input, input_state=conv_state, output_state=output_conv_state)
```

Ruff (0.16.0)

[error] Variable `input` is shadowing a Python builtin

(A001)